source/FreeRTOS_IPv6.c

```diff
@@ -421,7 +421,7 @@ eFrameProcessingResult_t eHandleIPv6ExtensionHeaders( NetworkBufferDescriptor_t
         }
     }
 
-    FreeRTOS_printf( ( "Extension headers : %s Truncated %u bytes. Removed %u, Payload %u xDataLength now %u\n",
+    FreeRTOS_printf( ( "Extension headers : %s Truncated %lu bytes. Removed %lu, Payload %u xDataLength now %lu\n",
                        ( eResult == eProcessBuffer ) ? "good" : "bad",
                        xMoveLen,
                        uxRemovedBytes,
```

Comment:
This alternative version will get less warning on (e.g. 64-bit) platforms:
~~~
    FreeRTOS_printf( ( "Extension headers : %s Truncated %u bytes. Removed %u, Payload %u xDataLength now %u\n",
                       ( eResult == eProcessBuffer ) ? "good" : "bad",
                       ( unsigned ) xMoveLen,
                       ( unsigned ) uxRemovedBytes,
                       FreeRTOS_ntohs( pxIPPacket_IPv6->xIPHeader.usPayloadLength ), // no cast needed for uint16_t
                       ( unsigned ) pxNetworkBuffer->xDataLength ) );
~~~
I'd like to have all printf formats for logging without the 'l' and other prefixes.